design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,195 @@
+    }
+
+    vault {
+      address = "http://vault.default:8200" # address of vault server
+      tls_skip_verify = "true"
+    }
+
+  config-init.hcl: |
+    pid_file = "/home/vault/pidfile"
+
+    auto_auth {
+        method "kubernetes" {
+            mount_path = "auth/kubernetes"
+            config = {
+                role = "rook-ceph"
+            }
+        }
+    }
+
+    cache {
+        use_auto_auth_token = true
+    }
+
+    listener "tcp" {
+        address = "127.0.0.1:8100" # port where vault agent listens in the pod
+        tls_disable = "true"
+    }
+
+    exit_after_auth = true
+
+    vault {
+      address = "http://vault.default:8200" # address of vault server
+      tls_skip_verify = "true"
+    }
+
+```
+
+### TLS authentication
+
+For TLS enabled vault server, the certs can be provided with help of k8s secret and added to `vault.hashicorp.com/tls-secret: <name of k8s secret>` and configure the `vault` block accordingly. The secret will be mounted in the path `/vault/tls/`.
+
+```yaml
+spec:
+  gateway:
+    annotations:
+      vault.hashicorp.com/agent-inject: 'true'
+      vault.hashicorp.com/agent-configmap: <ceph-object-store>-vault-agent-cm
+      vault.hashicorp.com/tls-secret: <name of k8s secret>
+
+---
+kind: ConfigMap
+metadata:
+  name: <ceph-object-store>-vault-agent-cm
+  namespace: rook-ceph
+apiVersion: v1
+data:
+  config.hcl: |
+
+  vault {
+      address = "http://vault.default:8200"
+      ca_cert = "/vault/tls/ca.crt"
+      client_cert = "/vault/tls/client.crt"
+      client_key = "/vault/tls/client.key"
+ }
+
+---
+kind: Secret
+metadata:
+  name: <name of k8s secret>
+  namespace: rook-ceph
+type: Opaque
+data:
+  client.key: <>
+  client.crt: <>
+  ca.crt:     <>
+
+```
+
+Or user can set it via following annotations instead of adding details in vault block:
+
+```yaml
+ annotations:
+   vault.hashicorp.com/ca-cert: '/vault/tls/ca.crt'
+   vault.hashicorp.com/client-cert: '/vault/tls/client.crt'
+   vault.hashicorp.com/client-key: '/vault/tls/client.key'
+   vault.hashicorp.com/tls-secret: '<name of k8s secret>'
+```
+
+Another option is user can set vault agent independently then set `VAULT_ADDR` to agent's address like below:
+
+!!! Note:
+Same configuration can be applied on `sse:s3` configuration
+
+```yaml
+security:
+  kms:
+    connectionDetails:
+        KMS_PROVIDER: vault
+        VAULT_ADDR: <address of vault agent>
+        VAULT_BACKEND_PATH: rook
+        VAULT_SECRET_ENGINE: kv
+        VAULT_AUTH_METHOD: kubernetes
+        VAULT_AUTH_KUBERNETES_ROLE: rook-ceph
+```
+
+## Config Commands
+
+The user can itself bring up `vault agent` as separate deployment configuring with service account authentication details. Then set following for RGW via toolbox pod:
+
+```bash
+ceph config set <ceph auth user for rgw> rgw_crypt_sse_s3_vault_addr <vault agent address>
+ceph config set <ceph auth user for rgw> rgw_crypt_sse_s3_vault_auth agent
+```
+
+## Another approach
```

Comment:
If we're not considering this approach, I'd suggest to remove it from the design doc. There are lots of alternate ways to implement, but we don't need to list them.